### ISSUE TITLE
[Question] Why couple mustache ?

### ISSUE THREAD

**Issue by bogas04** [felipeccastro/html-component#1]
I think it'll be better to let users work out their data-binding (querySelector + templateStrings, mustache, etc) instead of forcing mustache.

**Comment by felipeccastro** [comment]
Adding mustache was a simple way of working with single file components, since the template tag doesn't have any built-in data-binding method.
It is possible to use something else, like template strings for example, by simply overriding the `template` method on the component class. I'll change the README to better reflect that.

**Comment by bogas04** [comment]
Makes sense. I think what you can do is create a separate module `html-component-mustache-addon` and makes necessary bindings so that users don't end up downloading code that they won't use.

**Comment by ichpuchtli** [comment]
Would be cool to see jsx support as this can be strongly typed in typescript.

https://youtu.be/6wEVu_mkJjM?t=29m24s

![image](https://cloud.githubusercontent.com/assets/1134912/20293473/999bbe72-ab42-11e6-9a2a-e2fe3ba8639f.png)
![image](https://cloud.githubusercontent.com/assets/1134912/20293495/c64cc3d0-ab42-11e6-84de-39d367e2e415.png)
